test/functionalTest/cases/3272_custom_notification_sending_0-length_request/custom_notification_sending_0-length_request.test

```diff
@@ -0,0 +1,567 @@
+=================================================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+07. Create custom subscription with paylaod="{ %22x%22: 1}", includePayload=true
+================================================================================
+HTTP/1.1 201 Created
+Content-Length: 0
+Location: /v2/subscriptions/REGEX([0-9a-f]{24})
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+Date: REGEX(.*)
+
+
+
+08. Dump accumulator to see the notification
+============================================
+POST http://localhost:REGEX(\d+)/notify
+Fiware-Servicepath: /
+Content-Length: 129
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Ngsiv2-Attrsformat: normalized
+Host: localhost:REGEX(\d+)
+Accept: application/json
+Content-Type: application/json; charset=utf-8
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+
+{
+    "data": [
+        {
+            "A1": {
+                "metadata": {},
+                "type": "Boolean",
+                "value": true
+            },
+            "id": "E1",
+            "type": "T1"
+        }
+    ],
+    "subscriptionId": "REGEX([0-9a-f]{24})"
+}
+=======================================
+POST http://localhost:REGEX(\d+)/notify
+Fiware-Servicepath: /
+Content-Length: 129
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Ngsiv2-Attrsformat: normalized
+Host: localhost:REGEX(\d+)
+Accept: application/json
+Content-Type: application/json; charset=utf-8
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+
+{
+    "data": [
+        {
+            "A1": {
+                "metadata": {},
+                "type": "Boolean",
+                "value": true
+            },
+            "id": "E1",
+            "type": "T1"
+        }
+    ],
+    "subscriptionId": "REGEX([0-9a-f]{24})"
+}
+=======================================
+POST http://localhost:REGEX(\d+)/notify
+Fiware-Servicepath: /
+Content-Length: 9
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Ngsiv2-Attrsformat: custom
+Host: localhost:REGEX(\d+)
+Accept: application/json
+Content-Type: text/plain; charset=utf-8
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+
+{
+    "x": 1
+}
+=======================================
+POST http://localhost:REGEX(\d+)/notify
+Fiware-Servicepath: /
+Content-Length: 129
+User-Agent: orion/REGEX(\d+\.\d+\.\d+.*)
+Ngsiv2-Attrsformat: normalized
+Host: localhost:REGEX(\d+)
+Accept: application/json
+Content-Type: application/json; charset=utf-8
+Fiware-Correlator: REGEX([0-9a-f\-]{36})
+
+{
+    "data": [
+        {
+            "A1": {
+                "metadata": {},
+                "type": "Boolean",
+                "value": true
+            },
+            "id": "E1",
+            "type": "T1"
+        }
+    ],
+    "subscriptionId": "REGEX([0-9a-f]{24})"
+}
+=======================================
+
+
+09. GET the subscription
+========================
```

Comment:
Note that includedPayload field is not included in the JSON of the subscriptions in this step and it must be there. It should behave similar to other optional boolean fields in subscription, such as "onlyChanged".

Let me illustrate with an example that you can reproduce yourself. We create three subscriptions, each one with a different possibilities for onlyChanged: not included, set to true and set to false.

```
curl -v localhost:1026/v2/subscriptions -s -S -H 'Content-Type: application/json' -d @- <<EOF
{
  "description": "Without onlyChanges field",
  "subject": {
    "entities": [
      {
        "id": "Room1",
        "type": "Room"
      }
    ]
  },
  "notification": {
    "http": {
      "url": "http://localhost:1028/notify"
    }  
  }
}
EOF

curl -v localhost:1026/v2/subscriptions -s -S -H 'Content-Type: application/json' -d @- <<EOF
{
  "description": "With onlyChanges set to true",
  "subject": {
    "entities": [
      {
        "id": "Room1",
        "type": "Room"
      }
    ]
  },
  "notification": {
    "http": {
      "url": "http://localhost:1028/notify"
    },
    "onlyChangedAttrs": true
  }
}
EOF

curl -v localhost:1026/v2/subscriptions -s -S -H 'Content-Type: application/json' -d @- <<EOF
{
  "description": "With onlyChanges set to false",
  "subject": {
    "entities": [
      {
        "id": "Room1",
        "type": "Room"
      }
    ]
  },
  "notification": {
    "http": {
      "url": "http://localhost:1028/notify"
    },
    "onlyChangedAttrs": false
  }
}
EOF
```

now we get them and get the following:

```
$ curl -s -S  localhost:1026/v2/subscriptions | python -mjson.tool

[
    {
        "description": "Without onlyChanges field",
        "id": "5f7ee8b84f85b328ecc958ff",
        "notification": {
            "attrs": [],
            "attrsFormat": "normalized",
            "http": {
                "url": "http://localhost:1028/notify"
            },
            "onlyChangedAttrs": false
        },
        "status": "active",
        "subject": {
            "condition": {
                "attrs": []
            },
            "entities": [
                {
                    "id": "Room1",
                    "type": "Room"
                }
            ]
        }
    },
    {
        "description": "With onlyChanges set to true",
        "id": "5f7ee8b84f85b328ecc95900",
        "notification": {
            "attrs": [],
            "attrsFormat": "normalized",
            "http": {
                "url": "http://localhost:1028/notify"
            },
            "onlyChangedAttrs": true
        },
        "status": "active",
        "subject": {
            "condition": {
                "attrs": []
            },
            "entities": [
                {
                    "id": "Room1",
                    "type": "Room"
                }
            ]
        }
    },
    {
        "description": "With onlyChanges set to false",
        "id": "5f7ee8b94f85b328ecc95901",
        "notification": {
            "attrs": [],
            "attrsFormat": "normalized",
            "http": {
                "url": "http://localhost:1028/notify"
            },
            "onlyChangedAttrs": false
        },
        "status": "active",
        "subject": {
            "condition": {
                "attrs": []
            },
            "entities": [
                {
                    "id": "Room1",
                    "type": "Room"
                }
            ]
        }
    }
]
```

In summary:

* `"onlyChange": false` for the cases of ommitng and set to false
* `"onlyChange": true` for the case of set to true

The new field includePayload should behave similarly (with the difference that, in this case, omitting will result in true instead of false).